torch/distributed/_tensor/device_mesh.py

```diff
@@ -109,19 +109,21 @@ def __init__(
             if isinstance(mesh, torch.Tensor)
             else torch.tensor(mesh, dtype=torch.int)
         )
-        # always try to create default (world) pg, even if it is not initialized
-        # already. The world pg is used for device mesh identity (rank) on each
-        # process (we need to know if the current global rank is in the mesh or not)
-        self._get_or_create_default_group()
-        if _init_process_groups:
-            self._dim_groups = self._init_process_groups()
+        if self.mesh.numel() > 1:
+            # always try to create default (world) pg, even if it is not initialized
+            # already. The world pg is used for device mesh identity (rank) on each
+            # process (we need to know if the current global rank is in the mesh or not)
+            self._get_or_create_default_group()
+            if _init_process_groups:
+                self._dim_groups = self._init_process_groups()
 
     def _get_or_create_default_group(self):
         self._backend = Backend.GLOO if self.device_type == "cpu" else Backend.NCCL
         default_initialized = is_initialized()
+        if self.device_type == "ort":
```

Comment:
what does default_initialized mean?

Reply:
it means that the default process group is initialized. for ort, i'm requiring the script to initialize the default group before calling distribute_tensor.